library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -83,29 +87,34 @@ internal class TransactionPayloadFragment : Fragment(), TransactionFragment, Sea
 
     private fun populateUI() {
         if (isAdded && transaction != null) {
+            val isImage = transaction!!.responseContentType?.contains("image") ?: true
+            val imageData = if (isImage) transaction!!.byteData else null
             when (type) {
                 TYPE_REQUEST -> setText(
                     transaction!!.getRequestHeadersString(true),
                     transaction!!.getFormattedRequestBody(),
-                    transaction!!.isRequestBodyPlainText
+                    transaction!!.isRequestBodyPlainText,
+                    imageData
                 )
                 TYPE_RESPONSE -> setText(
                     transaction!!.getResponseHeadersString(true),
                     transaction!!.getFormattedResponseBody(),
-                    transaction!!.isResponseBodyPlainText
+                    transaction!!.isResponseBodyPlainText,
+                    imageData
                 )
             }
         }
     }
 
-    private fun setText(headersString: String, bodyString: String?, isPlainText: Boolean) {
+    private fun setText(headersString: String, bodyString: String?, isPlainText: Boolean, imageData: ByteArray?) {
```

Comment:
Can I ask you to rename this function to `setBody`?

Reply:
Done.